katdal/vis_flags_weights.py

```diff
@@ -39,25 +39,32 @@
 class VisFlagsWeights(object):
     """Correlator data in the form of visibilities, flags and weights.
 
+    This container stores the triplet of visibilities, flags and weights
+    as dask arrays to provide lazy / deferred access to the data.
+
     Parameters
     ----------
-    vis : array-like of complex64, shape (*T*, *F*, *B*)
+    vis : :class:`dask.array.Array` of complex64, shape (*T*, *F*, *B*)
         Complex visibility data as a function of time, frequency and baseline
-    flags : array-like of uint8, shape (*T*, *F*, *B*)
+    flags : :class:`dask.array.Array` of uint8, shape (*T*, *F*, *B*)
         Flags as a function of time, frequency and baseline
-    weights : array-like of float32, shape (*T*, *F*, *B*)
+    weights : :class:`dask.array.Array` of float32, shape (*T*, *F*, *B*)
         Visibility weights as a function of time, frequency and baseline
+    unscaled_weights : :class:`dask.array.Array` of float32, shape (*T*, *F*, *B*)
+        Weights that are not scaled by autocorrelations, thereby representing
+        the number of voltage samples that constitutes each visibility (optional)
     name : string, optional
         Identifier that describes the origin of the data (backend-specific)
 
     """
-    def __init__(self, vis, flags, weights, name='custom'):
+    def __init__(self, vis, flags, weights, unscaled_weights=None, name='custom'):
         if not (vis.shape == flags.shape == weights.shape):
             raise ValueError("Shapes of vis %s, flags %s and weights %s differ"
                              % (vis.shape, flags.shape, weights.shape))
```

Comment:
Should also check that `unscaled_weights` has the right shape if it is not None.

Reply:
Done.

Reply:
Done. Also added a test for this.